graphql_compiler/compiler/helpers.py

```diff
@@ -161,15 +161,47 @@ def validate_safe_string(value):
 
     if not set(value).issubset(VARIABLE_ALLOWED_CHARS) and \
             value not in legal_strings_with_special_chars:
-        raise GraphQLCompilationError(u'Encountered illegal characters in string: {}'.format(value))
+        raise GraphQLCompilationError(u'Encountered illegal characters in string: {}. Provided '
+                                      u'string is only allowed to have upper and lower case '
+                                      u'letters, digits and underscores.'
+                                      .format(value))
 
 
-def validate_output_name(value):
+def _validate_safe_argument_or_output_name(name, name_description):
+    """Ensure that the provided argument or output name does not have illegal characters."""
```

Comment:
I added a name description in order for the user to be able to quickly tell whether it was a tagged parameter, runtime argument or output name that failed validation. I think that it's important that this information is communicated. However, I am unsure of how to best reflect this change in the code. Here is one way to do it, but I am open to suggestions.